src/content-feature.js

```diff
@@ -424,15 +425,211 @@ export default class ContentFeature {
+                ...descriptor,
+                value: newFn
+            })
+        }
+
         return origDescriptor
     }
+
+    /**
+     * @template {keyof typeof globalThis} StandardInterfaceName
+     * @param {StandardInterfaceName} interfaceName - the name of the interface to shim (must be some known standard API, e.g. 'MediaSession')
+     * @param {typeof globalThis[StandardInterfaceName]} ImplClass - the class to use as the shim implementation
+     * @param {Partial<DefineInterfaceOptions>} [options] - options for defining the interface
+     */
+    shimInterface (
+        interfaceName,
+        ImplClass,
+        options
+    ) {
+        // TODO: validate that it does not exist already?
+
+        /** @type {DefineInterfaceOptions} */
+        const defaultOptions = {
+            allowConstructorCall: false,
+            disallowConstructor: false,
+            constructorErrorMessage: 'Illegal constructor',
+            interfaceDescriptorOptions: { writable: true, enumerable: false, configurable: true, value: ImplClass },
+            wrapToString: true
+        }
+
+        const fullOptions = { ...defaultOptions, ...options }
+
+        // In some cases we can get away without a full proxy, but in many cases below we need it.
+        // For example, we can't redefine `prototype` property on ES6 classes.
+        // Se we just always wrap the class to make the code more maintaibnable
+
+        /** @type {ProxyHandler<Function>} */
+        const proxyHandler = {}
+
+        // handle the case where the constructor is called without new
+        if (fullOptions.allowConstructorCall) {
+            // make the constructor function callable without new
+            proxyHandler.apply = function (target, thisArg, argumentsList) {
+                return Reflect.construct(target, argumentsList, target)
+            }
+        }
+
+        // make the constructor function throw when called without new
+        if (fullOptions.disallowConstructor) {
+            proxyHandler.construct = function () {
+                throw new TypeError(fullOptions.constructorErrorMessage)
+            }
+        }
+
+        if (fullOptions.wrapToString) {
+            // mask toString() on class methods. `ImplClass.prototype` is non-configurable: we can't override or proxy it, so we have to wrap each method individually
+            for (const [prop, descriptor] of objectEntries(getOwnPropertyDescriptors(ImplClass.prototype))) {
+                if (prop !== 'constructor' && descriptor.writable && typeof descriptor.value === 'function') {
+                    ImplClass.prototype[prop] = new Proxy(descriptor.value, toStringProxyMixin(descriptor.value, `function ${prop}() { [native code] }`))
+                }
+            }
+
+            // wrap toString on the constructor function itself
+            Object.assign(proxyHandler, toStringProxyMixin(ImplClass, `function ${interfaceName}() { [native code] }`))
```

Comment:
Here we're assign a `get` trap, I think make that clearer in the comment. (I can't think of a cleaner way to do this myself but the traps might collide is my concern).

Reply:
good point. I'll see what I can do